pkg/storage/pluginstore/grpc.go

```diff
@@ -374,12 +415,17 @@ func AsMapList(src []*structpb.Struct) []bson.M {
 	return dest
 }
 
+type converter func(src interface{}) interface{}
+
 // AsOrderedMap converts an array of protobuf structs into its original
 // representation, bson.D.
-func AsOrderedMap(src []*structpb.Struct) bson.D {
+func AsOrderedMap(src []*structpb.Struct, c ...converter) bson.D {
 	dest := make(bson.D, 0, len(src))
+	if c == nil {
+		c = []converter{ConvertFloatToInt}
+	}
```

Comment:
I don't think that it's possible for c to be nil here? If you call the function without passing any optional arguments the value is an empty array, not nil.

I noticed that in AsMap you always use the ConvertFloatToInt converter, even when they specify the optional custom converter parameter. Since you are passing c through to AsMap, I believe that this results in ConvertFloatToInt being used twice in AsMap when c is nil (which is never as I said above). 

I'm pretty sure that you can delete these lines here and it will work exactly the same.

```suggestion
	
```